Source/ARTTokenParams.h

```diff
@@ -14,10 +14,16 @@
 
 ART_ASSUME_NONNULL_BEGIN
 
+@protocol ARTDefaultTokenParams <NSObject>
```

Comment:
Hmm, I don't think this is a sound solution. This isn't really buying us anything, as we can in fact set `defaultTokenParams` to an `ARTTokenParams` with timestamp, and then we just force-convert `defaultTokenParams` back to `ARTTokenParams` ([here](https://github.com/ably/ably-ios/blob/master/Source/ARTAuth.m#L37)), so the timestamp would still be there! (Plus, that's generally a type-unsafe conversion.)

I don't think it really pays off to introduce a new type for this; you can just check on `defaultTokenParams`'s setter that `timestamp` is nil.

But if you want to introduce a new type to ensure this at compile time, I think `ARTDefaultTokenParams` shouldn't be a protocol but a class, with a public interface just like `ARTTokenParams` but without a `timestamp` property. That would ensure that we don't have a `timestamp` there, of course. It should have a method `toTokenParams` to be used in the line above that returns an `ARTTokenParams` with a nil `timestamp`. (Implementation suggestion: `ARTDefaultTokenParams` could have a `ARTTokenParams` _private_ property or instance variable to which it would just delegate all calls, and then return it when `toTokenParams` is called.)